torch/csrc/jit/codegen/cuda/test/test_gpu.cpp

```diff
@@ -9708,6 +9709,14 @@ TEST_F(NVFuserTest, FusionPersistentNormLocalShared_CUDA) {
   const float kEps = 1e-5;
   auto options = at::TensorOptions().dtype(at::kFloat).device(at::kCUDA, 0);
 
+  auto properties = at::cuda::getDeviceProperties(0);
+  const size_t required_smem_size =
+      (dimy - static_size) * sizeof(float) + TIDX * sizeof(float);
+  if (properties->sharedMemPerBlock < required_smem_size) {
```

Comment:
Question, what is the difference between `sharedMemPerBlock` and `sharedMemPerBlockOptin`? I didn't find the official documentation helpful:
```
size_t cudaDeviceProp::sharedMemPerBlock [inherited]
  Shared memory available per block in bytes

size_t cudaDeviceProp::sharedMemPerBlockOptin [inherited]
  Per device maximum shared memory per block usable by special opt in
```

Reply:
I didn't check thoroughly, but I assume the former is the default size, whereas the latter is the maximum size that can be configured. Since we don't do any smem size configuration, I think the former is the right limit.

Reply:
Is it related to this part of the code?https://github.com/csarofeen/pytorch/blob/a43cb20f48943595894e345865bc1eabf58a5b48/torch/csrc/jit/codegen/cuda/executor.cpp#L647-L676

Reply:
Oh, I didn't know we actually do configure the size when necessary.

https://github.com/csarofeen/pytorch/blob/a43cb20f48943595894e345865bc1eabf58a5b48/torch/csrc/jit/codegen/cuda/executor.cpp#L1207

I'll change the check to use `Optin`.

Reply:
FYI, those values on Titan RTX are: 

 ```
sharedMemPerBlock: 49152 
sharedMemPerBlockOptin: 65536
```